clang/test/SemaSYCL/unnamed-kernel.cpp

```diff
@@ -89,20 +96,29 @@ struct MyWrapper {
 
     using InvalidAlias1 = InvalidKernelName5;
 #ifndef __SYCL_UNNAMED_LAMBDA__
-    // expected-error@Inputs/sycl.hpp:220 {{kernel needs to have a globally-visible name}}
-    // expected-note@24 {{InvalidKernelName5 declared here}}
+    // expected-error@Inputs/sycl.hpp:220 {{'namespace1::KernelName<MyWrapper::InvalidKernelName5>' is an invalid kernel name type}}
+    // expected-note@Inputs/sycl.hpp:220 {{'MyWrapper::InvalidKernelName5' should be globally-visible}}
     // expected-note@+3{{in instantiation of function template specialization}}
 #endif
     q.submit([&](cl::sycl::handler &h) {
       h.single_task<namespace1::KernelName<InvalidAlias1>>([] {});
     });
+#ifndef __SYCL_UNNAMED_LAMBDA__
+    // expected-error@Inputs/sycl.hpp:220 {{'Templated_kernel_name2<Templated_kernel_name<InvalidKernelName1>>' is an invalid kernel name type}}
+    // expected-note@Inputs/sycl.hpp:220 {{'InvalidKernelName1' should be globally-visible}}
+    // expected-note@+3{{in instantiation of function template specialization}}
+#endif
+    q.submit([&](cl::sycl::handler &h) {
+      h.single_task<Templated_kernel_name2<Templated_kernel_name<InvalidKernelName1>>>([] {});
+    });
   }
 };
 
 int main() {
   cl::sycl::queue q;
 #ifndef __SYCL_UNNAMED_LAMBDA__
-  // expected-error@Inputs/sycl.hpp:220 {{kernel name is missing}}
+  // expected-error-re@Inputs/sycl.hpp:220 {{'(lambda at {{.*}}unnamed-kernel.cpp{{.*}}' is an invalid kernel name type}}
+  // expected-note@Inputs/sycl.hpp:220 {{kernel name is missing}}
```

Comment:
It would be great to update `kernel name is missing` to something like `Unnamed type cannot be used as kernel name` but it will be done as a separate change, right?

Reply:
Yes. @srividya-sundaram has already worked on this change in a separate patch and so I left this diagnostic as it is for now. She'll rebase and upload her patch once this is in